### ISSUE TITLE
ISC Customizations: Improvements to LTI xmodule

### ISSUE THREAD

**Issue by cewing** [jazkarta/edx-platform-for-isc#10]
Preserve the changes made in the following two commits:
- https://github.com/jazkarta/edx-platform/commit/ebcf93002557176102df2bf93ab79a979aaf0fba
- https://github.com/jazkarta/edx-platform/commit/8d771a5c1c16b9b82b1a5148f019a96e391f2c62

If possible, convert the `lti_spec_namespace` which is currently hard-coded to a configurable field that defaults to the value set in the original EdX code (http://www.imsglobal.org/services/ltiv1p1/xsd/imsoms_v1p0) and then use that field value when appropriate.  

Ensure that the tests added to cover this change are preserved and updated as required to make them work properly.

**Comment by cewing** [comment]
@natea: this seems like the type of thing that ought to be made a PR against EdX.  Making the LTI namespace configurable would allow the block to work with LTI Providers that don't use the "standard" namespace (like SCORMCloud).  Even in the light of a pending 'scorm-cloud' xblock, these changes would be useful (and are non-harmful to the existing implementation, I think).

**Comment by natea** [comment]
@cewing agreed.

**Comment by bryanlandia** [comment]
I've made a pull request to 'isc-birch-patches' branch of edx-platform and a related push to birch branch of ISCLC/ISC (edx configuration).  Have to get my head around how I want to do the PR to edx master, though.

**Comment by cewing** [comment]
@bryanlandia: I've commented on the PR, can you take a look there?  @natea: you too, please.

**Comment by natea** [comment]
@nedbat can you remind us how we could add a dictionary with coursewide parameters, so we don't have to override the LTI namespace directive to each instance of the LTI xmodule?

**Comment by cpennington** [comment]
@natea there's the `SettingsService`, which reads out of django settings. It's not per-course, though.

Otherwise, until we get something that functions as a course-scope, your best bet is probably to use an inherited field to allow the field to be set on the course, and then to flow down to the LTI xblock.

**Comment by cewing** [comment]
@cpennington, so this would be a field in the course settings, perhaps advanced settings or the like, which would be then inherited by blocks within the course?

**Comment by bryanlandia** [comment]
@natea Turns out `SettingsService` wasn't implemented for XBlocks until after the official Birch release.  Weird thing is that the release's version of `xblockutils` has mixins for `XBlock` that are set up to use `SettingsService`.  I'm going to do a bit more looking to see if we should just grab the new `xmodule.services` code with `SettingService` for our edx-platform 'fork' or go a different route.

**Comment by natea** [comment]
Thanks for the update Bryan. With the Cypress release just around the
corner, it might be best to wait for that since it'll have the
SettingsService for sure.

**Comment by bryanlandia** [comment]
@natea - OK, so put this one on hold?  Will the work on this remain part of the Jazkarta contract?

**Comment by bryanlandia** [comment]
PS  I do have a branch with what should be needed once `SettingsService` is ready to go.

**Comment by bryanlandia** [comment]
The code in the PR as is has been deployed to staging.  I'm placing this Issue on hold to revisit when we upgrade ISC to Cypress.  At that point we can look again at making the namespace spec URL configurable per course/XBlock instance.
